### PR TITLE
Address issue #3562 - InetAccessHandler should be able to only be applied to a specific set of connector names

### PR DESCRIPTION
**Pull request by nddipiazza**
Addresses #3562 

Allows you to specify a list of connector names that the ip access filter applies to.

This allows you to do things like have an open HTTPS connector but a whitelist restricted HTTP. 

Example:

`etc/jetty-inetaccess.xml`

```
<?xml version="1.0"?>
<!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "http://www.eclipse.org/jetty/configure_9_3.dtd">

<!-- =============================================================== -->
<!-- The IP Access Handler                                           -->
<!-- =============================================================== -->

<Configure id="Server" class="org.eclipse.jetty.server.Server">
  <Call name="insertHandler">
    <Arg>
      <New id="IPAccessHandler" class="org.eclipse.jetty.server.handler.IPAccessHandler">
  <Set name="connectorNames">
    <Array type="String">
	    <Item>http</Item>
	  </Array>
  </Set>
	<Set name="white">
	  <Array type="String">
	    <Item>127.0.0.1</Item>
	    <Item>127.0.1.1</Item>
	    <Item>192.168.1.229</Item>
	    <Item>192.168.1.199</Item>
	  </Array>
	</Set>
	<Set name="black">
	  <Array type="String">
	  </Array>
	</Set>
	<Set name="whiteListByPath">false</Set>
      </New>
    </Arg>
  </Call>
</Configure>

```

### CODE REVIEW

**Comment by joakime** [comment]
Assigning Handlers to specific Connectors is done with named connectors and virtualhosts.
Lets not invent yet another way to accomplish this.
If you are going to do this, don't use ports, use virtualhosts, like you see in `ContextHandler` now.

**Comment by nddipiazza** [comment]
@joakime Where I have been struggling with your answer from the email list is that I am using jetty via the `java -jar start.jar` only. Is it still possible to do what you are suggesting via a non-embedded Jetty startup? 

Do you have any examples that show how to set up a handler per-connector when you are running Jetty this way?

**Comment by gregw** [comment]
@nddipiazza with the `start.jar` style, for more custom configurations it may be necessary to replace an XML file (copy it from JETTY_HOME/etc to JETTY_BASE/etc and modify it there).

Now it may be that we still have to modify IPAccessHandler for your needs, but having a port black/white list doesn't feel right to me.   Having a black/white list of connector names is perhaps a better way to go?

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/InetAccessHandler.java [hunk 1]
```diff
@@ -123,12 +148,16 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
...
      * @return true if inetAddress and request are allowed
      */
-    protected boolean isAllowed(InetAddress address, HttpServletRequest request)
+    protected boolean isAllowed(InetAddress address, Request baseRequest, HttpServletRequest request)
     {
-        boolean allowed = _set.test(address);
+        String connectorName = baseRequest.getHttpChannel().getConnector().getName();
+
+        boolean allowed = _set.test(address) && (connectorName == null || _connectorNames.test(connectorName));
+        LOG.info("Connector name {} is included? {}", connectorName, _connectorNames.test(connectorName));
```
Comment (nddipiazza) [inline jetty-server/src/main/java/org/eclipse/jetty/server/handler/InetAccessHandler.java]: `connectorName` is null here.

**Comment by nddipiazza** [comment]
closing this - replacing with https://github.com/eclipse/jetty.project/pull/3572

**Pull request closed without merging**